src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,124 @@
+   * each page number button or link. Should use Anatomy Link or Button component.
+   */
+  paginationItem?: (page: number, isCurrent: boolean) => ReactElement<ButtonProps | LinkProps>;
+  onChange?: (page: number) => void;
+}
+
+const Pagination = ({
+  numberOfPages,
+  currentPage,
+  locale = 'en-US',
+  texts = {
+    nextAriaLabel: 'Next page',
+    previousAriaLabel: 'Previous page',
+    page: 'Page'
+  },
+  paginationItem,
+  onChange
+}: PaginationProps): JSX.Element => {
+  const bucketSize = 5;
+
+  const [pageNumbers, setPageNumbers] = useState<number[]>([]);
+  const [buckets, setBuckets] = useState<number[][]>(
+    Array.from(Array(Math.ceil(numberOfPages / bucketSize))).map(() => [])
+  );
+  const [currentPageNumber, setCurrentPageNumber] = useState<number>(currentPage || 1);
+
+  useEffect(() => {
+    const newBuckets: number[][] = Array.from(Array(Math.ceil(numberOfPages / bucketSize))).map(() => []);
+    for (let i = 1; i <= numberOfPages; i++) {
+      newBuckets[Math.ceil(i / bucketSize) - 1].push(i);
+    }
+    setBuckets(newBuckets);
+  }, [currentPageNumber, numberOfPages]);
+
+  useEffect(() => {
+    const bucketIndex = Math.ceil(currentPageNumber / bucketSize) - 1;
+    const numbers = buckets[bucketIndex];
+    if (numbers.length <= 2 && bucketIndex > 0) {
+      setPageNumbers(buckets[bucketIndex - 1].slice(numbers.length).concat(numbers));
+    } else {
+      setPageNumbers(numbers);
+    }
+  }, [buckets, currentPageNumber]);
+
+  useEffect(() => {
+    if (currentPage) {
+      setCurrentPageNumber(currentPage);
+    }
+  }, [currentPage]);
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(currentPageNumber);
+    }
+  }, [currentPageNumber, onChange]);
+
+  return (
+    <ul className="bsds-pagination">
+      {currentPageNumber !== 1 && (
+        <li className="bsds-pagination-stepper">
+          <Button
+            variant="subtle"
+            className="bsds-pagination-stepper-action"
+            aria-label={texts?.previousAriaLabel ?? 'Previous page'}
+            data-testid="prevBtn"
+            onClick={() => setCurrentPageNumber(currentPageNumber - 1)}
+          >
+            <IconChevronLeft className="bsds-icon-2x" />
+          </Button>
+        </li>
+      )}
+      {pageNumbers.map((number) => (
+        <li
+          key={number}
+          className={'bsds-pagination-page' + (number === currentPageNumber ? ' bsds-pagination-page-current' : '')}
+        >
+          {paginationItem ? (
+            paginationItem(number, number === currentPageNumber)
+          ) : (
+            <Button
+              variant="subtle"
+              aria-current={number !== currentPageNumber ? undefined : 'page'}
+              onClick={() => setCurrentPageNumber(number)}
+            >
+              <span className="bsds-visually-hidden">{texts?.page ?? 'Page '}</span>
+              {new Intl.NumberFormat(locale ?? 'en-US').format(number)}
```

Comment:
do we need to consider the order of the hidden text with the intl formatting too?

Reply:
We have the paginationItem renderer prop that could account for that situation. The only other way I could think to address this would be to aria-hide he number, and add the number in the visually-hidden span. with some kind of pattern to tell it where to place the number e.g. "Page #{number}"